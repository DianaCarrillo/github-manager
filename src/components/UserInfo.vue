/* eslint-disable no-labels */
<template>
  <section id="user-section">
    <div class="user">
      <img :src="user.avatarUrl" alt="" width="350" height="370px" />
      <Status :user="user" />
    </div>

    <p id="userName"> {{ user.name }}</p>
    <p id="userEmail"> {{ user.email }}</p>
    <p> Bio: {{ user.bio }}</p>
  </section>
</template>

<script>
// import User from './User.vue'
import gql from 'graphql-tag'

export default {
  name: 'Home',
  components: {
    // User,
  },
  data() {
    return {
      userName: 'DianaCarrillo',
      user: {},
    }
  },
  apollo: {
    user: {
      query: gql`
        query user($login: String!) {
          user(login: $login) {
            avatarUrl
            bio
            name
            email
          }
        }
      `,
      variables() {
        return {
          login: 'DianaCarrillo',
        }
      },
    },
  },
}
</script>

<style>
#user-section {
  position: relative;
  top: 30px;
  left: 100px;
}

#userName {
  font-size: 30px;
}
.user {
  border: black;
}
</style>
