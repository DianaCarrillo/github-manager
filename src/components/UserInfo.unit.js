import UserInfo from './UserInfo.vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'

describe('Displayed user correctly with query data', () => {
  it('displayd information quet data correctly', () => {
    const localVue = createLocalVue()
    const wrapper = shallowMount(UserInfo, { localVue })
    wrapper.setData({
      user: {
        avatarUrl: 'https://avatars0.githubusercontent.com/u/39560661?v=4',
        bio: 'Coder, psychologist, catlover',
        name: 'Diana',
        email: 'carrillorivera.diana@gmail.com',
        status: {
          id: '457524',
          message: 'Working',
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
