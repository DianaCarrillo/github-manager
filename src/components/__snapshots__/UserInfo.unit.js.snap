// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displayed user correctly with query data displayd information quet data correctly 1`] = `
<section
  id="user-section"
>
  <div
    class="user"
  >
    <img
      alt=""
      height="370px"
      src="https://avatars0.githubusercontent.com/u/39560661?v=4"
      width="350"
    />
     
    <status
      user="[object Object]"
    />
  </div>
   
  <p
    id="userName"
  >
     Diana
  </p>
   
  <p
    id="userEmail"
  >
     carrillorivera.diana@gmail.com
  </p>
   
  <p>
     Bio: Coder, psychologist, catlover
  </p>
</section>
`;
